vesc_hw_interface/src/vesc_hw_interface.cpp

```diff
@@ -182,19 +183,33 @@ CallbackReturn VescHwInterface::on_configure(const rclcpp_lifecycle::State& /*pr
     return CallbackReturn::FAILURE;
   }
 
-  if (command_mode_ == hardware_interface::HW_IF_POSITION)
+  if ((command_mode_ == hardware_interface::HW_IF_POSITION) || (command_mode_ == hardware_interface::HW_IF_VELOCITY) || (command_mode_ == hardware_interface::HW_IF_EFFORT))
+  {
+    vesc_interface_->requestMCConfiguration();
+    rclcpp::sleep_for(std::chrono::milliseconds(100));
+  }
+
+  upper_limit_ = 0.0;
+  lower_limit_ = 0.0;
+  homing_offset_ = 0.0;
+  homing_done_ = false;
+  homing_enabled_ = false;
+  if (command_mode_ == hardware_interface::HW_IF_POSITION || command_mode_ == "position_duty")
   {
-    auto upper_limit = 0.0;
-    auto lower_limit = 0.0;
     // parse URDF for limit parameters
     auto joint_limit_itr = info_.limits.find(joint_name_);
     if (joint_limit_itr != info_.limits.end())
     {
-      upper_limit = joint_limit_itr->second.max_position;
-      lower_limit = joint_limit_itr->second.min_position;
+      upper_limit_ = joint_limit_itr->second.max_position;
+      lower_limit_ = joint_limit_itr->second.min_position;
     } else {
       RCLCPP_WARN(rclcpp::get_logger("VescHwInterface"), "No joint position limits found in URDF, using default limits");
     }
+    homing_position_ = lower_limit_;
+    if (info_.hardware_parameters.find("servo/calibration_position") != info_.hardware_parameters.end())
+    {
+      homing_position_ = std::stod(info_.hardware_parameters["servo/calibration_position"]);
+    }
```

Comment:
If you're going to access it here, I think it would be better to share it with the servo controller via `init()` or a setter, just like the other variables.